pulsar-client-go/pulsar/client.go

```diff
@@ -101,17 +101,17 @@ type ClientOptions struct {
 type Client interface {
 	// Create the producer instance
 	// This method will block until the producer is created successfully
-	CreateProducer(ProducerOptions) (Producer, error)
+	CreateProducer(ProducerOptions, Schema) (Producer, error)
 
 	// Create a `Consumer` by subscribing to a topic.
 	//
 	// If the subscription does not exist, a new subscription will be created and all messages published after the
 	// creation will be retained until acknowledged, even if the consumer is not connected
-	Subscribe(ConsumerOptions) (Consumer, error)
+	Subscribe(ConsumerOptions, Schema) (Consumer, error)
 
 	// Create a Reader instance.
 	// This method will block until the reader is created successfully.
-	CreateReader(ReaderOptions) (Reader, error)
+	CreateReader(ReaderOptions, Schema) (Reader, error)
```

Comment:
For backward compatibility, can we do:

```
CreateReader(ReaderOptions) (Reader, error)

CreateReader(ReaderOptions, Schema) (Reader, error)
```

If we can't do that, I would do followings:

```
CreateReader(ReaderOptions) (Reader, error)

CreateReader(ReaderOptions, Schema) (Reader, error)
```